src/python/pants/core/util_rules/asdf.py

```diff
@@ -19,6 +21,27 @@
 logger = logging.getLogger(__name__)
 
 
+class AsdfPathString(str, Enum):
+    STANDARD = "<ASDF>"
+    LOCAL = "<ASDF_LOCAL>"
+
+    @staticmethod
+    def contains_strings(search_paths: Iterable[str]) -> tuple[bool, bool]:
+        return AsdfPathString.STANDARD in search_paths, AsdfPathString.LOCAL in search_paths
```

Comment:
Your type sig doesn't guarantee that this will work. You should wrap `search_paths` in a `set` before looping over it twice.